library/src/main/java/com/github/pwittchen/reactivenetwork/library/ReactiveNetwork.java

```diff
@@ -104,24 +104,54 @@ public ConnectivityStatus getConnectivityStatus(final Context context,
     }
 
     if (networkInfo.getType() == ConnectivityManager.TYPE_WIFI) {
-      if (checkInternet) {
-        return getWifiInternetStatus(networkInfo);
-      } else {
-        return ConnectivityStatus.WIFI_CONNECTED;
-      }
+      return ConnectivityStatus.WIFI_CONNECTED;
     } else if (networkInfo.getType() == ConnectivityManager.TYPE_MOBILE) {
       return ConnectivityStatus.MOBILE_CONNECTED;
     }
 
     return ConnectivityStatus.OFFLINE;
   }
 
-  private ConnectivityStatus getWifiInternetStatus(final NetworkInfo networkInfo) {
-    if (networkInfo.isConnected()) {
-      return ConnectivityStatus.WIFI_CONNECTED_HAS_INTERNET;
-    } else {
-      return ConnectivityStatus.WIFI_CONNECTED_HAS_NO_INTERNET;
-    }
+  /**
+   * Observes connectivity with the Internet with default settings. It pings remote host
+   * (www.google.com) at port 80 every 2 seconds with 2 seconds of timeout. This operation is used
+   * for determining if device is connected to the Internet or not. Please note that this method is
+   * less efficient than {@link #observeNetworkConnectivity(Context)} method and consumes data
+   * transfer, but it gives you actual information if device is connected to the Internet or not.
+   *
+   * @return RxJava Observable with Boolean - true, when we have an access to the Internet
+   * and false if not
+   */
+  public Observable<Boolean> observeInternetConnectivity() {
+    return observeInternetConnectivity(DEFAULT_PING_INTERVAL_IN_MS, DEFAULT_PING_HOST,
+        DEFAULT_PING_PORT, DEFAULT_PING_TIMEOUT_IN_MS);
+  }
+
+  /**
+   * Observes connectivity with the Internet by opening socket connection with remote host
+   *
+   * @param interval in milliseconds determining how often we want to check connectivity
+   * @param host for checking Internet connectivity
+   * @param port for checking Internet connectivity
+   * @param timeout for pinging remote host
+   * @return RxJava Observable with Boolean - true, when we have connection with host and false if
+   * not
+   */
+  public Observable<Boolean> observeInternetConnectivity(final int interval, final String host,
+      final int port, final int timeout) {
+    return Observable.interval(interval, TimeUnit.MILLISECONDS, Schedulers.io())
+        .map(new Func1<Long, Boolean>() {
+          @Override public Boolean call(Long tick) {
+            try {
+              Socket socket = new Socket();
+              socket.connect(new InetSocketAddress(host, port), timeout);
+              return socket.isConnected();
+            } catch (IOException e) {
+              return Boolean.FALSE;
+            }
+          }
+        })
+        .distinctUntilChanged();
```

Comment:
Great!  this looks like it will really help catch connection transitions when going underground.   Will start testing it at work, and over the weekend.